actix-web-codegen/src/route.rs

```diff
@@ -201,25 +241,47 @@ impl ToTokens for Route {
                     path,
                     guards,
                     wrappers,
+                    methods,
                 },
             resource_type,
         } = self;
         let resource_name = name.to_string();
-        let stream = quote! {
-            #[allow(non_camel_case_types, missing_docs)]
-            pub struct #name;
-
-            impl actix_web::dev::HttpServiceFactory for #name {
-                fn register(self, __config: &mut actix_web::dev::AppService) {
-                    #ast
-                    let __resource = actix_web::Resource::new(#path)
-                        .name(#resource_name)
-                        .guard(actix_web::guard::#guard())
-                        #(.guard(actix_web::guard::fn_guard(#guards)))*
-                        #(.wrap(#wrappers))*
-                        .#resource_type(#name);
-
-                    actix_web::dev::HttpServiceFactory::register(__resource, __config)
+        let stream = if guard != &GuardType::Multi {
```

Comment:
`if` branches differ in one line only. I suggest generating only that line inside `if`, and then use it as a part of the shared `quote`, like:

```rust
let guard = if guard != &GuardType::Multi {
    // ...
} else {
    // ...
};

let stream = quote! { ... };
```